cpu/stm32/periph/spi.c

```diff
@@ -64,6 +79,34 @@ static inline bool _use_dma(const spi_conf_t *conf)
 }
 #endif
 
+/**
+ * @brief Multiplier for clock divider calculations
+ *
+ * Makes the divider calculation fixed point
+ */
+#define SPI_APB_CLOCK_SHIFT          (0U)
+#define SPI_APB_CLOCK_MULT           (1U << SPI_APB_CLOCK_SHIFT)
+
+static uint8_t _get_clkdiv(const spi_conf_t *conf, uint32_t clock)
+{
+    uint32_t bus_clock = periph_apb_clk(conf->apbbus);
+    uint32_t div = (SPI_APB_CLOCK_MULT * bus_clock) / (2 * clock);
+    DEBUG("[spi] clock: divider: %"PRIu32"\n", div);
+    if (div <= SPI_APB_CLOCK_MULT) {
+        return 0;
+    }
+    uint8_t rounded_div = bitarithm_msb(div);
+    /* Determine if rounded_div is not a power of 2 */
+    if ((div & (div - 1)) != 0) {
+        /* increment by 1 to ensure that the clock speed at most the
+         * requested clock speed */
+        rounded_div++;
+    }
+    return rounded_div >= (BR_MAX + SPI_APB_CLOCK_SHIFT) ?
+                          BR_MAX :
+                          rounded_div - SPI_APB_CLOCK_SHIFT;
```

Comment:
This too - what is the relation to `SPI_APB_CLOCK_SHIFT` / `SPI_APB_CLOCK_MULT`?

Reply:
And simplified this. Hope it is a bit more clear what is happening now